### PR TITLE
support embeddings

### PR DESCRIPTION
**Pull request by jba**
The TaskType is optional.
A separate function is provided for a title.

### CODE REVIEW

File: genai/embed.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+package genai
+
+import (
+	"context"
+	"fmt"
+
+	pb "cloud.google.com/go/ai/generativelanguage/apiv1beta/generativelanguagepb"
+)
+
+func (c *Client) EmbeddingModel(name string) *EmbeddingModel {
```
Comment (eliben) [inline genai/embed.go:24]: Add a docstring here, specifically it should say that `name` is the name of the model used for embeddings

File: genai/embed.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	name     string
+	fullName string
+	TaskType TaskType
+}
+
+func (m *EmbeddingModel) EmbedContent(ctx context.Context, parts ...Part) (*EmbedContentResponse, error) {
+	return m.EmbedContentTitle(ctx, "", parts...)
+}
+
+func (m *EmbeddingModel) EmbedContentTitle(ctx context.Context, title string, parts ...Part) (*EmbedContentResponse, error) {
```
Comment (eliben) [inline genai/embed.go:45]: Nit: maybe `EmbedContentWithTitle` sounds a bit better? `EmbedContentTitle` sounds like we're embedding something that's a "content title" and it's not clear what that is.

Also add docstrings on these `Embed*` methods

File: genai/embed.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+		req.TaskType = &taskType
+	}
+	res, err := m.c.c.EmbedContent(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	return (EmbedContentResponse{}).fromProto(res), nil
+}
+
+// 	req := m.newEmbedContentRequest(parts,
```
Comment (eliben) [inline genai/embed.go:67]: Remove the commented-out code

**Review by eliben — APPROVED** [review]

File: genai/embed.go (hunk 1 above)
Comment (jba) [reply genai/embed.go:24]: done

File: genai/embed.go (hunk 2 above)
Comment (jba) [reply genai/embed.go:45]: Done.

**Pull request merged**